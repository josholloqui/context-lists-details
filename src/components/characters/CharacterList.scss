body {
  --darkcolor1: #292d3e;
  --darkcolor2: #d0d0d0;
  --darkcolor3: #f07178;
  --darkcolor4: #c3e88d;
  --darkcolor5: #ffcb6b;
  --darkcolor6: #82aaff;
  --darkcolor7: #c792ea;
  --darkcolor8: #89ddff;
  margin: 0;
}

#character-list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  grid-auto-rows: auto;
  list-style: none;
  text-align: center;
  justify-content: center;
  margin: 0;
  padding: 0;
}

.dark {
  color: #d0d0d0;
  background-color: #292d3e;
}

h1 {
  margin: 0;
  text-align: center;
}

header {
  text-align: center;
}
