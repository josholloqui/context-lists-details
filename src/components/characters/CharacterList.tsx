import * as React from 'react';
import CharacterItem from './CharacterItem';
import { useCharacters } from '../../hooks/characters';

const CharacterList: React.FunctionComponent = () => {
  const characters = useCharacters();

  const characterElements = characters.map(character => (
    <li key={character.id}>
      <CharacterItem {...character} />
    </li>
  ));

  return (
    <ul>
      {characterElements}
    </ul>
  );
};

export default CharacterList;
