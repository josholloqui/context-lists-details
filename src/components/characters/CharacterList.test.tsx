import * as React from 'react';
import { render, cleanup } from '@testing-library/react'
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterList component', () => {
  afterEach(() => cleanup());

  it('render CharacterList', async() => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList />
      </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
