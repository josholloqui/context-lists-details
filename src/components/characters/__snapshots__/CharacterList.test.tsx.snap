// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component render CharacterList 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
